agg-ast/schema_derivation/src/lib.rs

```diff
@@ -418,9 +418,18 @@ pub(crate) fn insert_required_key_into_document(
 pub(crate) fn remove_field(schema: &mut Schema, path: Vec<String>) {
     if let Some((field, field_path)) = path.split_last() {
         let input = get_schema_for_path_mut(schema, field_path.into());
-        if let Some(Schema::Document(d)) = input {
-            d.keys.remove(field);
-            d.required.remove(field);
+        match input {
+            Some(Schema::Document(d)) => {
+                d.keys.remove(field);
+                d.required.remove(field);
+            }
+            Some(Schema::Array(a)) => {
+                if let Schema::Document(d) = a.as_mut() {
+                    d.keys.remove(field);
+                    d.required.remove(field);
+                }
+            }
```

Comment:
```suggestion
            Some(Schema::Array(Schema::Document(d))) => {
                d.keys.remove(field);
                d.required.remove(field);
            }
```
Is this legal rust? Could be nice to condense this directly into the pattern as opposed to using an `if` inside the pattern branch.

Reply:
That would be legal, however, array types are wrapped in a `Box` which isn't possible to pattern match